### PR TITLE
Moving to the new dev environment code structure

### PR DESCRIPTION
**Pull request by dimpar**
In this PR we're moving existing random-beacon Solidity related code from `/solidity` dir to `/solidity-v1` which becomes our "old" version of random-beacon. The v2 version of random-beacon and ecdsa will be developed in `/solidity/random-beacon` & `/solidity/ecdsa` respectively.
All the existing references to `/solidity` dir were also replaced with `solidity-v1`

### CODE REVIEW

**Comment by pdyraga** [comment]
Would it be possible to split this PR into two: one for moving `solidity` to `solidity-v2` and doing all the necessary changes in GitHub Actions and another PR setting up a new dev environment in `solidity/random-beacon`? I think that would greatly help in reviews.

**Comment by pdyraga** [comment]
Side note, it is good that `mv` is one commit (8d1bdc9) with no additional changes in the content of files - the commit history for all files is preserved.

**Comment by dimpar** [comment]
> Would it be possible to split this PR into two: one for moving `solidity` to `solidity-v2` and doing all the necessary changes in GitHub Actions and another PR setting up a new dev environment in `solidity/random-beacon`? I think that would greatly help in reviews.

Sure, the first part "moving `solidity` to `solidity-v2` and doing all the necessary changes in GitHub Actions" is done in this PR. The second part will be handled in a seperate PR.

File: .github/workflows/contracts.yml
Comment (nkuba) [inline .github/workflows/contracts.yml:7]: What if we rename the file to `contracts-v1.yml`?
cc: @michalinacienciala

File: .github/workflows/contracts.yml
Comment (nkuba) [inline .github/workflows/contracts.yml:261]: This will require an update in https://github.com/keep-network/ci/blob/main/config/config.json as well.

File: .github/workflows/dashboard-format.yaml
Comment (nkuba) [inline .github/workflows/dashboard-format.yaml:11]: Let's also update it above:
```
on:
  push:
    branches:
      - main
    paths:
      - "solidity-v1/dashboard/**"
```

File: .github/workflows/dashboard-testnet.yml
Comment (nkuba) [inline .github/workflows/dashboard-testnet.yml:38]: Let's update it also above:
```
  push:
    branches:
     - main
    paths:
      - "solidity-v1/dashboard/**"
```

File: .github/workflows/docs.yml
Comment (nkuba) [inline .github/workflows/docs.yml:87]: Let's update filters on top of the file:
```
  push:
    branches:
      - main
    paths:
      - "docs/**"
      - "solidity-v1/**"
      - "!solidity-v1/dashboard/**"
```

File: .github/workflows/genesis-testnet.yml
Comment (nkuba) [inline .github/workflows/genesis-testnet.yml:17]: `cache-dependency-path` below also has to be updated

File: .github/workflows/npm.yml
Comment (nkuba) [inline .github/workflows/npm.yml:14]: Paths filter on top:
```
      - 'solidity-v1/contracts/**'
      - 'solidity-v1/package.json'
      - 'solidity-v1/package-lock.json'
```

**Review by nkuba — CHANGES_REQUESTED** [review]
We need to update `client.yml` and `dashboard-mainnet.yml`.

File: .github/workflows/contracts.yml
Comment (dimpar) [reply .github/workflows/contracts.yml:7]: I think it can be expanded to other `*.yml` files as well like `dashboard-*.yml` to stay consistent with the naming convention. Besides, we can also rename the names of the workflows at line 1 in each file where applicable. 
Btw. there's a nice feature request in GitHub to make [workflows run from subfolders](https://github.community/t/feature-request-allow-workflow-configuration-in-sub-folders/16776)

File: .github/workflows/contracts.yml
Comment (dimpar) [reply .github/workflows/contracts.yml:261]: yea, good point https://github.com/keep-network/ci/pull/21

File: .github/workflows/dashboard-format.yaml
Comment (dimpar) [reply .github/workflows/dashboard-format.yaml:11]: bf8f2de

File: .github/workflows/docs.yml
Comment (dimpar) [reply .github/workflows/docs.yml:87]: bf8f2de

File: .github/workflows/genesis-testnet.yml
Comment (dimpar) [reply .github/workflows/genesis-testnet.yml:17]: bf8f2de

File: .github/workflows/npm.yml
Comment (dimpar) [reply .github/workflows/npm.yml:14]: bf8f2de

**Review by michalinacienciala — CHANGES_REQUESTED** [review]
Left two notes.

Apart from that, I think we also need to update:
https://github.com/keep-network/keep-core/blob/main/.dockerignore
https://github.com/keep-network/keep-core/blob/main/README.adoc

And maybe (I'm not sure):
https://github.com/keep-network/keep-core/blob/main/crytic-config.json
/https://github.com/keep-network/keep-core/blob/main/infrastructure/eth-networks/keep-test/ropsten/KeepRandomBeaconOperator.json
https://github.com/keep-network/keep-core/blob/main/infrastructure/eth-networks/keep-test/ropsten/KeepToken.json
https://github.com/keep-network/keep-core/blob/main/infrastructure/eth-networks/keep-test/ropsten/TokenStaking.json

Also, there are places in `local-setup` repo where `keep-core/solidity/` is being referenced. We should create PR updating those references as well.

File: .github/workflows/contracts.yml
Comment (dimpar) [reply .github/workflows/contracts.yml:7]: Renamed 5 files here: 12b4c4c

Regarding the dashboard code being in `solidity-v1` we have plans to move the dashboard to a new home https://github.com/threshold-network/token-dashboard

File: .github/workflows/dashboard-testnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-testnet.yml:98]: e704d91 and other missed references

**Comment by dimpar** [comment]
Related PRs:
- https://github.com/keep-network/ci/pull/21
- https://github.com/keep-network/keep-ecdsa/pull/916
- https://github.com/keep-network/coverage-pools/pull/197

File: .github/workflows/dashboard-format-v1.yaml
Comment (nkuba) [inline .github/workflows/dashboard-format-v1.yaml:1]: Thoughts about naming the files: `dashboard-v1*.yaml` instead of `dashboard-*-v1.yaml`, i.e.:
```
dashboard-v1-format.yaml
dashboard-v1-mainnet.yaml
dashboard-v1-testnet.yaml
```

Comment (dimpar) [reply .github/workflows/dashboard-format-v1.yaml]: 0f607aa

File: .github/workflows/dashboard-v1-mainnet.yml
Comment (nkuba) [inline .github/workflows/dashboard-v1-mainnet.yml:1]:
```suggestion
name: Token Dashboard v1 / Mainnet
```

File: .github/workflows/dashboard-v1-format.yaml
Comment (nkuba) [inline .github/workflows/dashboard-v1-format.yaml:1]:
```suggestion
name: Token Dashboard v1 / Code Format Checks
```

File: .github/workflows/dashboard-v1-testnet.yml
Comment (nkuba) [inline .github/workflows/dashboard-v1-testnet.yml:1]:
```suggestion
name: Token Dashboard v1 / Testnet
```

**Review by nkuba — APPROVED** [review]

**Comment by nkuba** [comment]
LGTM, I'm leaving the final approval decision to @michalinacienciala

File: .github/workflows/dashboard-v1-testnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-v1-testnet.yml]: c2dec9d

File: .github/workflows/dashboard-v1-format.yaml
Comment (dimpar) [reply .github/workflows/dashboard-v1-format.yaml]: c2dec9d

File: .github/workflows/dashboard-v1-mainnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-v1-mainnet.yml]: c2dec9d

**Review by nkuba — APPROVED** [review]

File: .gitignore
Comment (pdyraga) [inline .gitignore:2]: This file no longer exists, could be a good opportunity to clean up.

File: README.adoc
Comment (pdyraga) [inline README.adoc:49]: Can we explain what is the difference between `solidity-v1` and `solidity`?

**Review by pdyraga — COMMENTED** [review]
Heads up @r-czajkowski @michalsmiarowski, you may run into conflicts with this PR.

File: .gitignore
Comment (dimpar) [reply .gitignore]: 0d674f7

File: .github/workflows/dashboard-v1-testnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-v1-testnet.yml]: ok, `v1` was put there to easily spot the difference, but I guess "T Token Dashboard" is fine too. c93ce5a

File: .github/workflows/dashboard-v1-mainnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-v1-mainnet.yml]: c93ce5a

File: .dockerignore
Comment (dimpar) [reply .dockerignore:24]: cd7e012 I've added the same structure for `solidity/random-beacon`

File: .dockerignore
Comment (pdyraga) [inline .dockerignore:27]: `package.json` or `package-lock.json`?

**Comment by pdyraga** [comment]
It looks good to me but I'll hold on with merging until we create a release branch for the v1.15 dashboard.

File: .github/workflows/dashboard-testnet.yml
Comment (michalinacienciala) [reply .github/workflows/dashboard-testnet.yml:98]: One more thing to update in the README.adoc: change `solidity.yml` to `solidity-v1.yml` in [line 3](https://github.com/keep-network/keep-core/blame/main/README.adoc#L3).

**Review by michalinacienciala — CHANGES_REQUESTED** [review]
Left one small [comment]()https://github.com/keep-network/keep-core/pull/2610#discussion_r717935735, also there are conflicts that need to be resolved.
Once we fix that, I'm ok with merging.

File: .github/workflows/dashboard-testnet.yml
Comment (dimpar) [reply .github/workflows/dashboard-testnet.yml]: 36d6cbe

**Comment by dimpar** [comment]
There were a lot of conflicts between the code that was moved to `/solidity-v1` and the new release changes for the keep dashboard. We also had `solidity/random-beacon` dir that already contained new code for beacon v2. The easiest approach to resolve the conflicts was to branch off of `main` and `cherry-pick + squash` the relevant commits. Below 2 commits should contain every change that was done previously:
- ef8cbbc - moving existing beacon code to the `/solidity-v1`
- 3c1fa1d - replacing references `solidity -> solidity-v1`

@pdyraga @michalinacienciala @nkuba ready to verify if it looks good

**Review by nkuba — APPROVED** [review]

**Review by pdyraga — APPROVED** [review]

**Review by michalinacienciala — APPROVED** [review]

File: README.adoc
Comment (dimpar) [reply README.adoc:79]: Sure, let's do this in a separate PR. The longer we wait for the merge, the more conflicts it will create.

**Pull request merged**
